tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/jsf/JSFModuleProjectLocationValidationService.java

```diff
@@ -0,0 +1,154 @@
+/*******************************************************************************
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ *
+ *******************************************************************************/
+package com.liferay.ide.project.core.jsf;
+
+import java.io.File;
+
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.sapphire.FilteredListener;
+import org.eclipse.sapphire.Listener;
+import org.eclipse.sapphire.PropertyContentEvent;
+import org.eclipse.sapphire.modeling.Path;
+import org.eclipse.sapphire.modeling.Status;
+import org.eclipse.sapphire.services.ValidationService;
+
+
+/**
+ * @author Simon Jiang
+ */
+public class JSFModuleProjectLocationValidationService extends ValidationService
+{
+    private Listener listener;
+
+    private boolean canCreate( File file )
+    {
+        while( !file.exists() )
+        {
+            file = file.getParentFile();
+
+            if( file == null )
+            {
+                return false;
+            }
+        }
+
+        return file.canWrite();
+    }
+
+    @Override
+    protected Status compute()
```

Comment:
IS the logic in this method similar to other validation logic in other wizards?  can any of that be shared instead of re-written here?